lms/djangoapps/courseware/date_summary.py

```diff
@@ -240,13 +256,45 @@ def is_enabled(self):
 
     @lazy
     def date(self):
+        deadline = None
+
         try:
-            verified_mode = CourseMode.objects.get(
-                course_id=self.course_id, mode_slug=CourseMode.VERIFIED
-            )
-            return verified_mode.expiration_datetime
+            verified_mode = CourseMode.objects.get(course_id=self.course_id, mode_slug=CourseMode.VERIFIED)
+            deadline = verified_mode.expiration_datetime
         except CourseMode.DoesNotExist:
-            return None
+            pass
+
+        if self.course and self.course_overview.self_paced and self.enrollment:
+            # Check if the given course has opted out of the feature
+            course_config = CourseDynamicUpgradeDeadlineConfiguration.current(self.course.id)
+
+            # NOTE: We check the pk here to determine if we have an instance generated from the current() method or
+            # pulled from the database. If the instance isn't from the database, it is not relevant.
```

Comment:
My original implementation was along these lines. It was the early `return` (or lack thereof) that tripped me up originally. Thanks!